hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/s3guard/DynamoDBMetadataStore.java

```diff
@@ -377,11 +377,12 @@ private DynamoDB createDynamoDB(
    * FS via {@link S3AFileSystem#shareCredentials(String)}; this will
    * increment the reference counter of these credentials.
    * @param fs {@code S3AFileSystem} associated with the MetadataStore
+   * @param ttlTimeProvider
    * @throws IOException on a failure
    */
   @Override
   @Retries.OnceRaw
-  public void initialize(FileSystem fs) throws IOException {
+  public void initialize(FileSystem fs, ITtlTimeProvider ttlTimeProvider) throws IOException {
```

Comment:
line width

Reply:
**I've made the FileSystem TTL Provider one of the attributes you can get from the FS via a StoreContext*

bindToOwnerFilesystem() is picking this up, looks like all that is needed is to delete the `timeProvider = new S3Guard.TtlTimeProvider(conf);` line for the owner binding to be picked up

Reply:
Discussed offline with Gabor. Outcome of that conversation: bindToOwnerFileSystem doesn't exist everywhere and there isn't already a context created outside of the context (ha!) of certain operations. But we should have a context created earlier since it doesn't contain state that changes between operations (I actually wonder why we're creating a new instance for every operation instead of the metadatastore getting a permanent context). We need to check the context is complete enough, as this is called during FS initialization, precisely when the createStoreContext() javadoc warns you to be careful :)

Reply:
I think we should avoid that for now, and use it during another refactor maybe. I've fixed all the other stuff with my latest commit. Excluding this would change the whole point of this jira/pr

Reply:
OK. Just be aware, I consider it transient.

@mackrorysd -as to why the context changes: it includes the metastore, so once that is created things changed. But yes, everything in init is brittle; including stuff like setting up delegation tokens (needed before you make any S3 calls)